library/network_connections.py

```diff
@@ -2265,51 +2245,17 @@ def _try_reapply(self, idx, con):
 
     def run_action_down(self, idx):
         connection = self.connections[idx]
-
-        cons = self.nmutil.connection_list(name=connection["name"])
-        changed = False
-        if cons:
-            seen = set()
-            while True:
-                ac = Util.first(
-                    self.nmutil.active_connection_list(
-                        connections=cons, black_list=seen
-                    )
-                )
-                if ac is None:
-                    break
-                seen.add(ac)
-                self.log_info(
-                    idx, "down connection %s: %s" % (connection["name"], ac.get_path())
-                )
-                changed = True
-                self.connections_data_set_changed(idx)
-                if self.check_mode == CheckMode.REAL_RUN:
-                    try:
-                        self.nmutil.active_connection_deactivate(ac)
-                    except MyError as e:
-                        self.log_error(idx, "down connection failed: %s" % (e))
-
-                    wait_time = connection["wait"]
-                    if wait_time is None:
-                        wait_time = 10
-
-                    try:
-                        self.nmutil.active_connection_deactivate_wait(ac, wait_time)
-                    except MyError as e:
-                        self.log_error(
-                            idx, "down connection failed while waiting: %s" % (e)
-                        )
-
-                cons = self.nmutil.connection_list(name=connection["name"])
-        if not changed:
-            message = "down connection %s failed: connection not found" % (
-                connection["name"]
-            )
-            if connection[PERSISTENT_STATE] == ABSENT_STATE:
-                self.log_info(idx, message)
-            else:
-                self.log_error(idx, message)
+        logger = logging.getLogger()
+        log_handler = NmLogHandler(self.log, idx)
+        logger.addHandler(log_handler)
```

Comment:
maybe there should be a `with self.logger(idx)` which returns a context manager for the logger.